### ISSUE TITLE
flops_percentage unaccurate

### ISSUE THREAD

**Issue by PatGlan** [hikvision-research/SAViT#1]
Hello,
I just tried to reimplemented your Method, but it turns out, that the resulting total pruning-ratio of 91% ("total percentage: 91.200"), although I set the parameter "pruning_flops_percentage=0.5". Do you know, why the resulting-prune-ratio is that appart from the goal of 50%?
What do I have to change to end up with the goal-ratio?

Here are my trainings-settings:
I used all parameter as described in the readme for your pruning-procedure, except that I changed the dataloader.
For simplicity I replaced the dataloader of Imagenet by the standard Dataloader of CIFAR100 of pytorch, but rescaled the image to size 224. Since, I haven't got a pretrained model for CIFAR100, I did not used a pretrained model (flag: --fine_tune).

Can you help me with that problem?

best regards
Patrick

**Comment by zuttoanemone** [comment]
> Hello, I just tried to reimplemented your Method, but it turns out, that the resulting total pruning-ratio of 91% ("total percentage: 91.200"), although I set the parameter "pruning_flops_percentage=0.5". Do you know, why the resulting-prune-ratio is that appart from the goal of 50%? What do I have to change to end up with the goal-ratio?
> 
> Here are my trainings-settings: I used all parameter as described in the readme for your pruning-procedure, except that I changed the dataloader. For simplicity I replaced the dataloader of Imagenet by the standard Dataloader of CIFAR100 of pytorch, but rescaled the image to size 224. Since, I haven't got a pretrained model for CIFAR100, I did not used a pretrained model (flag: --fine_tune).
> 
> Can you help me with that problem?
> 
> best regards Patrick



> Hello, I just tried to reimplemented your Method, but it turns out, that the resulting total pruning-ratio of 91% ("total percentage: 91.200"), although I set the parameter "pruning_flops_percentage=0.5". Do you know, why the resulting-prune-ratio is that appart from the goal of 50%? What do I have to change to end up with the goal-ratio?
> 
> Here are my trainings-settings: I used all parameter as described in the readme for your pruning-procedure, except that I changed the dataloader. For simplicity I replaced the dataloader of Imagenet by the standard Dataloader of CIFAR100 of pytorch, but rescaled the image to size 224. Since, I haven't got a pretrained model for CIFAR100, I did not used a pretrained model (flag: --fine_tune).
> 
> Can you help me with that problem?
> 
> best regards Patrick

Did u solve the problem at last? I met the exact same issue as u occurred.I thought there's some logic problem in main.py or some of the module it invoked.I compared my log and theirs and its quite obvious that the program didn't finish one epoch before it trys to mask the neural node.
By the way,instead of using the ImageNet101k i use other minor datasets and processed it into the form it takes,just as u,i think so.

Also,even i use the worng masked model to deploy,the amount of calculation didn't get sharpen correctly,the model gets smaller thought.

apologize for my poor english.
